From: Bill MacAllister <bill@ca-zephyr.org>
Date: Fri, 15 May 2020 18:32:56 -0700
Subject: Remove Makefile rules that conflict with Debian

This patch is from the work of Spencer Olson <olsonse@umich.edu>.  The
patches make builds possible using Debian package building tools.
---
 Makefile | 92 +---------------------------------------------------------------
 1 file changed, 1 insertion(+), 91 deletions(-)

diff --git a/Makefile b/Makefile
index ad67936..c12357d 100644
--- a/Makefile
+++ b/Makefile
@@ -18,8 +18,6 @@ LNS		:= ln -sf
 #
 ###############################################################################
 VERSION		:= $(word 2,$(shell grep "^Version:" $(SPECFILE)))
-TARBALL		:= kafs-client-$(VERSION).tar
-ZTARBALL	:= $(TARBALL).bz2
 
 ###############################################################################
 #
@@ -42,34 +40,7 @@ endif
 #
 ###############################################################################
 all:
-	$(MAKE) -C src all
-
-###############################################################################
-#
-# Install everything
-#
-###############################################################################
-MAN1	:= $(MANDIR)/man1
-MAN5	:= $(MANDIR)/man5
-MAN7	:= $(MANDIR)/man7
-MAN8	:= $(MANDIR)/man8
-
-install: all
-	$(MAKE) -C src install
-	$(MKDIR) -p -m755 $(DESTDIR)$(MAN1)/
-	$(MKDIR) -p -m755 $(DESTDIR)$(MAN5)/
-	$(MKDIR) -p -m755 $(DESTDIR)$(MAN7)/
-	$(MKDIR) -p -m755 $(DESTDIR)$(MAN8)/
-	$(INSTALL) -D -m 0644 man/*.1 $(DESTDIR)$(MAN1)/
-	$(INSTALL) -D -m 0644 man/*.5 $(DESTDIR)$(MAN5)/
-	$(INSTALL) -D -m 0644 man/*.7 $(DESTDIR)$(MAN7)/
-	$(INSTALL) -D -m 0644 man/*.8 $(DESTDIR)$(MAN8)/
-	$(INSTALL) -D -m 0644 conf/cellservdb.conf $(DESTDIR)$(DATADIR)/cellservdb.conf
-	$(INSTALL) -D -m 0644 conf/etc.conf $(DESTDIR)$(ETCDIR)/kafs/client.conf
-	$(INSTALL) -D -m 0644 conf/kafs_dns.conf $(DESTDIR)$(ETCDIR)/request-key.d/kafs_dns.conf
-	$(INSTALL) -D -m 0644 conf/kafs-config.service $(DESTDIR)$(UNITDIR)/kafs-config.service
-	$(INSTALL) -D -m 0644 conf/afs.mount $(DESTDIR)$(UNITDIR)/afs.mount
-	$(MKDIR) -m755 $(DESTDIR)$(ETCDIR)/kafs/client.d
+	$(MAKE) -C src progs
 
 ###############################################################################
 #
@@ -82,64 +53,3 @@ clean:
 
 distclean: clean
 	$(MAKE) -C src distclean
-	$(RM) -r rpmbuild $(TARBALL)
-
-###############################################################################
-#
-# Generate a tarball
-#
-###############################################################################
-$(ZTARBALL):
-	git archive --prefix=kafs-client-$(VERSION)/ --format tar HEAD | \
-	bzip2 -9 >$(ZTARBALL)
-
-tarball: $(ZTARBALL)
-
-###############################################################################
-#
-# Generate an RPM
-#
-###############################################################################
-SRCBALL	:= rpmbuild/SOURCES/$(TARBALL)
-ZSRCBALL := rpmbuild/SOURCES/$(ZTARBALL)
-
-BUILDID	:= .local
-dist	:= $(word 2,$(shell grep -r "^%dist" /etc/rpm /usr/lib/rpm))
-release3:= $(word 2,$(shell grep ^Release: $(SPECFILE)))
-release2:= $(subst %{?dist},$(dist),$(release3))
-release1:= $(subst %{?buildid},$(BUILDID),$(release2))
-release	:= $(subst %{?distprefix},,$(release1))
-rpmver	:= $(VERSION)-$(release)
-SRPM	:= rpmbuild/SRPMS/kafs-client-$(rpmver).src.rpm
-
-RPMBUILDDIRS := \
-	--define "_srcrpmdir $(CURDIR)/rpmbuild/SRPMS" \
-	--define "_rpmdir $(CURDIR)/rpmbuild/RPMS" \
-	--define "_sourcedir $(CURDIR)/rpmbuild/SOURCES" \
-	--define "_specdir $(CURDIR)/rpmbuild/SPECS" \
-	--define "_builddir $(CURDIR)/rpmbuild/BUILD" \
-	--define "_buildrootdir $(CURDIR)/rpmbuild/BUILDROOT"
-
-RPMFLAGS := \
-	--define "buildid $(BUILDID)"
-
-rpm: tarball
-	mkdir -p rpmbuild
-	chmod ug-s rpmbuild
-	mkdir -p rpmbuild/{SPECS,SOURCES,BUILD,BUILDROOT,RPMS,SRPMS}
-	cp $(ZTARBALL) $(ZSRCBALL)
-	rpmbuild -ts $(ZSRCBALL) --define "_srcrpmdir rpmbuild/SRPMS" $(RPMFLAGS)
-	rpmbuild --rebuild $(SRPM) $(RPMBUILDDIRS) $(RPMFLAGS)
-
-rpmlint: rpm
-	rpmlint $(SRPM) $(CURDIR)/rpmbuild/RPMS/*/kafs-client-{,debuginfo-}$(rpmver).*.rpm
-
-###############################################################################
-#
-# Build debugging
-#
-###############################################################################
-show_vars:
-	@echo VERSION=$(VERSION)
-	@echo TARBALL=$(TARBALL)
-	@echo BUILDFOR=$(BUILDFOR)
